diff --git a/config/optimize b/config/optimize
index 91ce8283d4..c9466a3185 100644
--- a/config/optimize
+++ b/config/optimize
@@ -9,7 +9,15 @@ TARGET_LIBDIR="$SYSROOT_PREFIX/lib $SYSROOT_PREFIX/usr/lib"
 TARGET_INCDIR="$SYSROOT_PREFIX/include $SYSROOT_PREFIX/usr/include"
 
 HOST_CPPFLAGS=""
-HOST_CFLAGS="-march=x86-64 -mtune=generic -O2 -Wall -pipe -I$TOOLCHAIN/include"
+case "$(uname -m)" in
+  x86_64)
+    HOST_CPU_CFLAGS="-march=x86-64 -mtune=generic"
+    ;;
+  *)
+    HOST_CPU_CFLAGS=""
+    ;;
+esac
+HOST_CFLAGS="${HOST_CPU_CFLAGS} -O2 -Wall -pipe -I$TOOLCHAIN/include"
 HOST_CXXFLAGS="$HOST_CFLAGS"
 HOST_LDFLAGS="-Wl,-rpath,$TOOLCHAIN/lib -L$TOOLCHAIN/lib"
 HOST_INCDIR="$TOOLCHAIN/include /usr/include"
